lib/puppet/util/posix.rb

```diff
@@ -12,9 +45,16 @@ module Puppet::Util::POSIX
   class << self
     # Returns an array of all the groups that the user's a member of.
     def groups_of(user)
-      groups = []
-      Puppet::Etc.group do |group|
-        groups << group.name if group.mem.include?(user)
+      begin
+        require 'puppet/ffi/posix'
+        groups = GrpLibC.user_groups(user)
```

Comment:
I wonder, can we lose the `GrpLibC` namespace and rename the `user_groups` method to `getgrouplist`, so it's more obvious how it does things? I know the method does more than getgrouplist, but I think it'd be more clear that way.